lib/v2/index.js

```diff
@@ -0,0 +1,144 @@
+const { Datatypes, errors, extractDomain } = require("../utils");
+const Config = require("./config");
+const { Crypto, Http, Forms, Input } = require("./core");
+const { deriveSharedSecret } = require("../utils/crypto");
+
+class EvervaultClient {
+  constructor(teamId, appId, customConfig = {}) {
+    if (!Datatypes.isString(teamId)) {
+      throw new errors.InitializationError("teamId must be a string");
+    }
+    if (!Datatypes.isString(appId)) {
```

Comment:
Should we validate that app id looks correct? Not being able to do so for teamId is annoying because it is leading to the keys issue when the teamId is invalid